### PR TITLE
feat: introduce a lightweight workflow which utilizes composite actions

### PR DESCRIPTION
**Pull request by awownysz-splunk**
This PR introduces a lightweight workflow to get faster feedback on PRs. The new approach utilizes composite actions to allow for separate definition of each job.

Based on the following MSCS TA Jira story: [https://splunk.atlassian.net/browse/ADDON-66448?atlOrigin=eyJpIjoiYjc1NzdkNzc4MTlkNDM0ODg2M2EzZDY0YzgyNGZmMjMiLCJwIjoiaiJ9](url)

Sample run of the reused workflow in the MSCS TA repo: [https://github.com/splunk/splunk-add-on-for-microsoft-cloud-services/actions/workflows/lightweight-check-pr.yml](https://github.com/splunk/splunk-add-on-for-microsoft-cloud-services/actions/workflows/lightweight-check-pr.yml)

### CODE REVIEW

**Comment by awownysz-splunk** [comment]
I have read the CLA Document and I hereby accept the CLA

**Review by artemrys — COMMENTED** [review]
@awownysz-splunk Are you planning to reuse those actions in the main reusable workflow eventually?

**Comment by awownysz-splunk** [comment]
> @awownysz-splunk Are you planning to reuse those actions in the main reusable workflow eventually?

I guess I can do this in one go - I'll update the PR then

**Comment by awownysz-splunk** [comment]
@artemrys `reusable-build-test-release.yml` is now refactored using Composite Actions that I've created for the lightweight workflow. You can see a successful workflow run here: [](https://github.com/splunk/splunk-add-on-for-microsoft-cloud-services/actions/runs/7103546362)
Jobs for UI tests, etc. remain unchanged - I haven't implemented separate actions for them as they were not needed for the lightweight workflow. I guess refactoring this part would be a separate task.

File: .github/actions/unit-tests/action.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+      run: |
+        if [ -f "poetry.lock" ]
+          then
+            mkdir -p package/lib || true
+            pip install poetry==1.5.1 poetry-plugin-export==1.4.0
+            poetry export --without-hashes -o package/lib/requirements.txt
+            poetry export --without-hashes --dev -o requirements_dev.txt
+          fi
+        if [ ! -f requirements_dev.txt ]; then echo no requirements;exit 0 ;fi
+        git config --global url."https://${{ inputs.GH_TOKEN_ADMIN }}@github.com".insteadOf https://github.com
```
Comment (awownysz-splunk) [inline .github/actions/unit-tests/action.yml:37]: This command is needed to properly install dependencies from internal repos. Discovered while working on [https://splunk.atlassian.net/browse/ADDON-66730?atlOrigin=eyJpIjoiMDdlM2ZhMTE4NDMwNDViZDkyOGNhZWRmYzdiZjcwYzIiLCJwIjoiaiJ9](https://splunk.atlassian.net/browse/ADDON-66730?atlOrigin=eyJpIjoiMDdlM2ZhMTE4NDMwNDViZDkyOGNhZWRmYzdiZjcwYzIiLCJwIjoiaiJ9)

Comment (artemrys) [reply .github/actions/unit-tests/action.yml:35]: https://github.com/srv-rr-github-token needs to be added to the repository with at least `read` access?

Comment (awownysz-splunk) [reply .github/actions/unit-tests/action.yml:35]: From what I just tested - https://github.com/srv-rr-github-token doesn't have to have explicit access to the repo. Having an internal repo - created as a Splunk org - should be enough.

**Comment by artemrys** [comment]
I am changing the PR to be a "feat" so we can have a release and you can properly reference it in your repo.

**Review by artemrys — APPROVED** [review]
Good job!

**Pull request merged**